### PR TITLE
Pil 728: RFM- content and uk based pages

### PR DESCRIPTION
**Pull request by alikouhestani**

### CODE REVIEW

File: app/views/rfm/CheckNewFilingMemberView.scala.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    layout: templates.Layout,
+    govukButton: GovukButton
+)
+
+@(mode:Mode)(implicit request: Request[_], appConfig: FrontendAppConfig, messages: Messages)
+
+
+@layout(pageTitle = titleNoForm(messages("checkNewFilingMember.title"))) {
+
+    <span class="govuk-caption-l">@messages("checkNewFilingMember.heading.caption")</span>
```
Comment (jassalrichy) [inline app/views/rfm/CheckNewFilingMemberView.scala.html:32]: update sectionHeading component:
`<h2 class="govuk-caption-l hmrc-caption-l"><span id="section-header" class="govuk-caption-l">@messages</span></h2>`
and use here instead:
`<span class="govuk-caption-l">@messages("checkNewFilingMember.heading.caption")</span>`

File: app/views/rfm/CheckNewFilingMemberView.scala.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <h1 class="govuk-heading-l">@messages("checkNewFilingMember.heading")</h1>
```
Comment (jassalrichy) [inline app/views/rfm/CheckNewFilingMemberView.scala.html:33]: use heading component:
`@heading(messages("checkNewFilingMember.heading"), "govuk-heading-l")`

File: app/views/rfm/CheckNewFilingMemberView.scala.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    <p class="govuk-body">@messages("checkNewFilingMember.p1")</p>
```
Comment (jassalrichy) [inline app/views/rfm/CheckNewFilingMemberView.scala.html:35]: use paragraphBody component:
`@paragraphBody(messages("checkNewFilingMember.p1"))`

File: app/views/rfm/NFMRegisteredInUKConfirmationView.scala.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+@layout(pageTitle = title(form, messages("isNFMRegisteredInUK.title"))) {
+
+    @formHelper(action = controllers.rfm.routes.UkBasedFilingMemberController.onSubmit(mode), 'autoComplete -> "off") {
+
+        @if(form.errors.nonEmpty) {
+            @govukErrorSummary(ErrorSummaryViewModel(form,  errorLinkOverrides = Map("value" -> "value_0")))
+        }
+
+        <span class="govuk-caption-l">@messages("isNFMRegisteredInUK.heading.caption")</span>
```
Comment (jassalrichy) [inline app/views/rfm/NFMRegisteredInUKConfirmationView.scala.html:39]: use updated sectionHeading component (see above) and move outside of form:
`@sectionHeader(messages("isNFMRegisteredInUK.heading.caption"))`

File: app/views/rfm/NFMRegisteredInUKConfirmationView.scala.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
+@*
```
Comment (jassalrichy) [inline app/views/rfm/NFMRegisteredInUKConfirmationView.scala.html:1]: rename this view to 'UkBasedFilingMemberView.scala.html' inlione with it's controller name.

File: app/controllers/rfm/UkBasedFilingMemberController.scala [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  def onSubmit(mode: Mode): Action[AnyContent] = (rfmIdentify andThen getData andThen requireData).async { implicit request =>
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
+        ukBasedFilingMember =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(NfmRegisteredInUKPage, ukBasedFilingMember))
```
Comment (jassalrichy) [inline app/controllers/rfm/UkBasedFilingMemberController.scala:69]: Should we not be updating '**RfmUkBasedPage**' ?

File: app/controllers/rfm/UkBasedFilingMemberController.scala [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  getData:                   DataRetrievalAction,
+  requireData:               DataRequiredAction,
+  formProvider:              NFMRegisteredInUKConfirmationFormProvider,
+  val controllerComponents:  MessagesControllerComponents,
+  view:                      NFMRegisteredInUKConfirmationView
+)(implicit ec:               ExecutionContext, appConfig: FrontendAppConfig)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  val form = formProvider()
```
Comment (jassalrichy) [inline app/controllers/rfm/UkBasedFilingMemberController.scala:47]: add type annotation:
`val form: Form[Boolean] = formProvider()`

File: app/views/rfm/CheckNewFilingMemberView.scala.html (hunk 1 above)
Comment (alikouhestani) [reply app/views/rfm/CheckNewFilingMemberView.scala.html:32]: Not sure what you mean here can we connect and discuss this

File: app/views/rfm/CheckNewFilingMemberView.scala.html (hunk 2 above)
Comment (alikouhestani) [reply app/views/rfm/CheckNewFilingMemberView.scala.html:33]: Done

File: app/views/rfm/CheckNewFilingMemberView.scala.html (hunk 3 above)
Comment (alikouhestani) [reply app/views/rfm/CheckNewFilingMemberView.scala.html:35]: Done

File: app/views/rfm/NFMRegisteredInUKConfirmationView.scala.html (hunk 5 above)
Comment (alikouhestani) [reply app/views/rfm/NFMRegisteredInUKConfirmationView.scala.html:1]: Agreed.Done

File: app/controllers/rfm/UkBasedFilingMemberController.scala (hunk 6 above)
Comment (alikouhestani) [reply app/controllers/rfm/UkBasedFilingMemberController.scala:69]: Nice one. Done

File: app/views/rfm/CheckNewFilingMemberView.scala.html (hunk 1 above)
Comment (alikouhestani) [reply app/views/rfm/CheckNewFilingMemberView.scala.html:32]: Surely just using @sectionHeader is sufficient?
@jassalrichy

**Review by jassalrichy — DISMISSED** [review]

File: app/views/rfm/UkBasedFilingMemberView.scala.html [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    govukRadios: GovukRadios,
+    sectionHeader: sectionHeader,
+    govukButton: GovukButton
+)
+
+
+@(form: Form[_], mode: Mode)(implicit request: Request[_], appConfig: FrontendAppConfig, messages: Messages)
+
+@layout(pageTitle = title(form, messages("isNFMRegisteredInUK.title"))) {
+    @sectionHeader(messages("isNFMRegisteredInUK.heading.caption"))
```
Comment (jassalrichy) [inline app/views/rfm/UkBasedFilingMemberView.scala.html:34]: @alikouhestani Can we check if placing the sectionHeader outside the form causes a rendering issue if the govukErrorSummary is displayed ?

**Review by jassalrichy — APPROVED** [review]

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**